cpp/src_prims/linalg/reduce_rows_by_key.h

```diff
@@ -342,7 +342,6 @@ void sum_rows_by_key_large_nkeys_rowmajor( const DataIteratorT d_A,
   grid.z = std::min(grid.z, MAX_BLOCKS);
   //std::cout << "block = " << block.x << ", " << block.y << std::endl;
   //std::cout << "grid = " << grid.x << ", " << grid.y << ", " << grid.z << std::endl;
-  cudaMemsetAsync(d_sums, 0, sizeof(DataType)*nkeys*ncols, st);
```

Comment:
You'd still need this memset, right?

Reply:
No, we don't need this here. The `cudaMemsetAsync` call in `reduce_rows_by_key` method at line 379 takes care of the memset functionality.  
In fact, this call here was another bug which was always resetting the first 1024 entries to zero when `nkeys > 1024`.

Reply:
ah.. makes sense. Thanks Kamesh.